### ISSUE TITLE
Diffuser

### ISSUE THREAD

**Issue by pman193** [0x5e/homebridge-tuya-platform#411]
### Detail Informations

Im not a coder. I own a diffuser compatible with Tuya app. In app I can change the brightness and the color including changing it from a static color to an RGB gradient and nightlight feature. Im using Homebridge to add this to HomeKit. Any way to add the color functionality to your plugin? I want to be able to choose the color in HomeKit and also switch from gradient to fixed color.

**Comment by pman193** [comment]
Hopefully, someone will see this. This is my devices DP Instructions. It's an Essential Oils Diffuser with a Led Light. The light can be set to gradient, fixed or nightlight. This plugin will only control brightness for the lights. Can't pick a color or change it to fixed color. Control for the mist is wonky.  it shuts off when I slide the slider all the way up in HomeKit. Any thoughts.  Im not a coder. Thanks in advance.

Power	Boolean	
"{true,false}"

Light	Boolean	
"{true,false}"

countdown	Enum	
{
  "range": [
    "0",
    "1",
    "2",
    "3"
  ]
}

dp_mist_grade	Enum	
{
  "range": [
    "small",
    "big",
    "off"
  ]
}

colour_data	String	
{
  "maxlen": 255
}

work_mode	Enum	
{
  "range": [
    "white",
    "colour",
    "scene",
    "scene1",
    "scene2",
    "scene3",
    "scene4"
  ]
}

lightmode	Enum	
{
  "range": [
    "1",
    "2",
    "3"
  ]
}

setlight	Integer	
{
  "unit": "",
  "min": 0,
  "max": 255,
  "scale": 0,
  "step": 1
}

**Comment by pman193** [comment]
@0x5e See above

**Comment by 0x5e** [comment]
https://github.com/0x5e/homebridge-tuya-platform/issues/432#issuecomment-1886242662
@pman193 You can try this, not sure if it's working.

**Comment by pman193** [comment]
@0x5e I added this to the plugin config json and still doesn't allow color picking or setting light to fixed or gradient. Not a coder and im copy and pasting changes. I notice my diffusers light shows as a string and something about maxlen: 255. Also, on the Tuya developer site it shows its category as jsq instead of xxj. Thanks for all your help so far. its appreciated.

**Comment by pman193** [comment]
@0x5e Yeah. None of these tuya plugins properly support diffusers. The LED light on my Asakuki only turns on gradient. And you can't properly control the mist either. Thank you but your work on this.

**Comment by pman193** [comment]
@0x5e Hello again! Thanks again or all your hard work in creating this plugin. I noticed while on the Tuya Developer site that my particular oil diffuser is categorized as jsq and not xxj. I believe this is the challenge you referred to. Is there any way to change / edit that on the Tuya site?
